product/roundhouse/infrastructure.app/ConfigurationPropertyHolder.cs

```diff
@@ -40,6 +40,8 @@ public interface ConfigurationPropertyHolder
         string VersionTableName { get; set; }
         string ScriptsRunTableName { get; set; }
         string ScriptsRunErrorsTableName { get; set; }
+        [Obsolete("Use EnvironmentNames")]
+        string EnvironmentName { get; set; }

```

Comment:
This may be able to be set from the RoundHousE api, so best to check if you can property config in first

Reply:
I didn't quite understand what you mean. I have effectively just reverted to "old" (as in original EnvironmentName -> EnvironmentNames PR) behaviour. Should we make "EnvironmentName" a Facade for EnvironmentNames instead, maybe? (i.e. write-through on set, read from EnvironmentNames on read?)

Reply:
I think EnvironmentName is effectively a facade now

Reply:
it could probably use a little bit more to make that change effective.

Reply:
Yes, maybe. So we might as well just make it a proper facade in `ConfigurationPropertyHolder`, and move the "if empty then the other one, else this" logic there, or remove it altogether (as in the original trigger-happy PR). What do you prefer? 

I prefer making `EnvironmentNames` a list (of some sorts) of string, and make conversion (split on comma or whatever) on reading command-line. It makes more sense when working with RoundhousE as a library. Unbound strings are evil. You can store anything from a single character to the complete works of Shakespeare in them, and there's no way of telling from the signature if there is a "hidden format" you should use (e.g. comma-separated strings).

Reply:
I prefer one environment name, but it should be converted (and I thought the original PR adding this handled the splitting?) to a list.

Reply:
There may be a reason the others are all strings here though. Check to see if there is a list type in property configuration holder - if not, there may be a reason for it.

Reply:
I will have a closer look.

Reply:
Agree that the property should be a string collection and that it should be split when the command line is read. Parsing a comma separated string provided as an API argument is just silly.

We can have the obsoleted singular property that is just a facade for the collection; agree with that approach.

All of this code is present, just a matter of some reorganization. I'm busy tonight, but I can definitely put this together Wednesday night. If you want to do this one, feel free, but I definitely don't mind jumping in on some code (build systems bore me to tears...yes, they're really important, but I'd rather watch a wall of paint dry.)

Reply:
Lovely if you can have a look, @BiggerNoise. I'd feel more comfortable on this one if we have more than two eyes on it, as we've had some discussions around it. Can you technically commit to/continue on the same pull request, how does that work?

Reply:
Perfect. I Agree, if we have many people working on the same thing, a branch is better. But not necessary here.